WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductVariationFormViewModel.swift

```diff
@@ -0,0 +1,215 @@
+    private(set) var actionsFactory: ProductFormActionsFactoryProtocol
+
+    /// Not applicable to product variation form
+    private(set) var password: String? = nil
+
+    /// Not applicable to product variation form
+    private(set) var productName: Observable<String>? = nil
+
+    private let productVariationSubject: PublishSubject<ProductVariation> = PublishSubject<ProductVariation>()
+    private let isUpdateEnabledSubject: PublishSubject<Bool>
+
+    /// The product variation before any potential edits; reset after a remote update.
+    private var originalProductVariation: ProductVariation {
+        didSet {
+            productVariation = originalProductVariation
+        }
+    }
+
+    /// The product variation with potential edits; reset after a remote update.
+    private var productVariation: ProductVariation {
+        didSet {
+            guard productVariation != oldValue else {
+                return
+            }
+
+            actionsFactory = ProductVariationFormActionsFactory(productVariation: productVariation)
+            productVariationSubject.send(productVariation)
+            isUpdateEnabledSubject.send(hasUnsavedChanges())
+        }
+    }
+
+    private let productImageActionHandler: ProductImageActionHandler
+    private let storesManager: StoresManager
+    private var cancellable: ObservationToken?
+
+    init(productVariation: ProductVariation,
+         productImageActionHandler: ProductImageActionHandler,
+         storesManager: StoresManager = ServiceLocator.stores) {
+        self.productImageActionHandler = productImageActionHandler
+        self.storesManager = storesManager
+        self.originalProductVariation = productVariation
+        self.productVariation = productVariation
+        self.actionsFactory = ProductVariationFormActionsFactory(productVariation: productVariation)
+        self.isUpdateEnabledSubject = PublishSubject<Bool>()
+        self.cancellable = productImageActionHandler.addUpdateObserver(self) { [weak self] allStatuses in
+            if allStatuses.productImageStatuses.hasPendingUpload {
+                self?.isUpdateEnabledSubject.send(true)
+            }
+        }
+    }
+
+    deinit {
+        cancellable?.cancel()
+    }
+
+    func hasUnsavedChanges() -> Bool {
+        return productVariation != originalProductVariation || productImageActionHandler.productImageStatuses.hasPendingUpload
+    }
+
+    func hasProductChanged() -> Bool {
+        return productVariation != originalProductVariation
+    }
+
+    func hasPasswordChanged() -> Bool {
+        // no-op
+        return false
+    }
+}
+
+// MARK: - More menu
+//
+extension ProductVariationFormViewModel {
+    func canEditProductSettings() -> Bool {
+        return false
+    }
+
+    func canViewProductInStore() -> Bool {
+        // no-op
+        return false
+    }
+}
+
+// MARK: Action handling
+//
+extension ProductVariationFormViewModel {
+    func updateName(_ name: String) {
+        // no-op
```

Comment:
I would add that, explicitly, we don't want to set the name because a product variation doesn't have it, and the same for all the fields that we don't need to update. 👍

Reply:
sounds good! updated here: https://github.com/woocommerce/woocommerce-ios/pull/2578/commits/7afaed7b2ad39821c4c5714e78d677058f7c0e35